### ISSUE TITLE
When the refresh data is [], the page will be refreshed by clicking/drag

### ISSUE THREAD

**Issue by glhfwp** [ankeetmaini/react-infinite-scroll-component#188]
When the refresh data is [], the page will be refreshed by clicking/drag/drop...

https://codesandbox.io/s/falling-architecture-gb1zt?file=/src/index.js
```jsx
import React from "react";
import { render } from "react-dom";
import InfiniteScroll from "react-infinite-scroll-component";

const style = {
  height: 30,
  border: "1px solid green",
  margin: 6,
  padding: 8
};

class App extends React.Component {
  state = {
    items: Array.from({ length: 20 }),
    hasMore: false, // test PullRequest
    loading: false
  };
  ajaxData = refresh => {
    // a fake async api call like which sends
    // 20 more records in .5 secs
    if (this.state.items.length >= 100) {
      this.setState({ hasMore: false, loading: false });
      return;
    }
    setTimeout(() => {
      refresh === "refresh" && this.reset();
      this.setState({
        loading: false,
        // items: this.state.items.concat(Array.from({ length: 20 }))
        items: this.state.items.concat([]) // BUG
      });
    }, 1000);
  };
  fetchMoreData = (refresh = "") => {
    this.setState(
      {
        loading: true
      },
      () => this.ajaxData(refresh)
    );
  };
  refresh = () => {
    this.fetchMoreData("refresh");
  };
  reset = () => {
    this.setState({
      items: []
    });
  };

  render() {
    return (
      <div>
        <h1>demo: react-infinite-scroll-component</h1>
        <h2>{this.state.loading ? "Loading..." : ""}</h2>
        <hr />
        <InfiniteScroll
          dataLength={this.state.items.length}
          next={this.fetchMoreData}
          hasMore={this.state.hasMore}
          loader={<h4>Loading...</h4>}
          endMessage={
            <p style={{ textAlign: "center" }}>
              <b>Yay! You have seen it all</b>
            </p>
          }
          pullDownToRefresh
          refreshFunction={this.refresh}
          pullDownToRefreshContent={
            <h3 style={{ textAlign: "center" }}>
              &#8595; Pull down to refresh
            </h3>
          }
          releaseToRefreshContent={
            <h3 style={{ textAlign: "center" }}>&#8593; Release to refresh</h3>
          }
        >
          {this.state.items.map((i, index) => (
            <div style={style} key={index}>
              div - #{index}
            </div>
          ))}
        </InfiniteScroll>
      </div>
    );
  }
}

render(<App />, document.getElementById("root"));

```

**Comment by xahshengx** [comment]
same issue, seems like after pull down to refresh, it remains at release to request. Hence triggering call for every click/touch

**Comment by xahshengx** [comment]
This is because componentWillReceiveProps will not update pullToRefreshThresholdBreached state to false if data.length is not changed.

Made a pull request, but got a feeling nobody's looking at it.

#189

**Comment by geoffreyteng** [comment]
fix for this has been merged and released #190

**Comment by glhfwp** [comment]
Thanks so much
